### PR TITLE
feat: API to run Renamer-CLI binaries

### PR DESCRIPTION
**Pull request by Deepak-png981**
Created the API but I need to figure a way to implement it using  the dynamic binaries ( solved )
Updated the API endpoint to run the Linux Binaries

### CODE REVIEW

File: src/app/api/runBinary/route.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        const { stdout, stderr } = await execFilePromise(binaryPath, args, {
+            env: {
+                ...process.env,
+                OPENAI_API_KEY: process.env.OPENAI_API_KEY,
+            }
+        });
+        console.log('stdout:', stdout);
+
+        if (stderr) {
+            console.error('Binary error:', stderr);
```
Comment (deepak-joshi-cs-coder) [inline src/app/api/runBinary/route.ts:117]: Major: If the binary generates stderr, the process continues without proper interruption or error response.

File: src/app/api/runBinary/route.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        const jsonData = fs.readFileSync('output.json', 'utf8');
+        const renamedFiles = JSON.parse(jsonData);
+
+        return NextResponse.json({ renamedFiles });
+
+    } catch (error) {
+        console.error('Error processing file:', error);
+        return NextResponse.json({ error: 'Failed to process the file.' }, { status: 500 });
+    } finally {
```
Comment (deepak-joshi-cs-coder) [inline src/app/api/runBinary/route.ts:132]: Are their chances of getting a Race condition , Cleanup in finally may fail if the file is still being accessed.

File: src/app/api/runBinary/route.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        if (!fs.existsSync(paths.tempDir)) {
+            fs.mkdirSync(paths.tempDir);
+        }
+
+        paths.tempFilePath = path.join(paths.tempDir, fileName);
+
+        fs.writeFileSync(paths.tempFilePath, fileContent, 'utf8');
+
+
+        const binaryPath = 'E:\\Random Project\\renamer-app\\src\\app\\api\\runBinary\\renamer-win.exe';
```
Comment (deepak-joshi-cs-coder) [inline src/app/api/runBinary/route.ts]: we are planning to get this from the Renamer repo.

File: src/app/api/runBinary/route.ts (hunk 2 above)
Comment (Deepak-png981) [reply src/app/api/runBinary/route.ts:132]: I think it is fine for now.

**Pull request merged**